conda_smithy/configure_feedstock.py

```diff
@@ -580,7 +583,96 @@ def _load_forge_config(forge_dir, variant_config_files):
     return config
 
 
-def main(forge_file_directory, variant_config_files):
+def get_installed_version(pkg):
+    from conda_build.conda_interface import root_dir, linked
+    root_linked = linked(root_dir)
+    vers_inst = [dist.split('::', 1)[-1].rsplit('-', 2)[1] for dist in root_linked
+        if dist.split('::', 1)[-1].rsplit('-', 2)[0] == pkg]
+    if not len(vers_inst) == 1:
+        return None
+    return vers_inst[0]
+
+
+def check_version_uptodate(resolve, name, installed_version, error_on_warn):
+    from conda_build.conda_interface import VersionOrder, MatchSpec
+    available_versions = [pkg.version for pkg in resolve.get_pkgs(MatchSpec(name))]
+    available_versions = sorted(available_versions, key=VersionOrder)
+    most_recent_version = available_versions[-1]
+    if installed_version is None:
+        msg = "{} is not installed in root env.".format(name)
+    elif VersionOrder(installed_version) < VersionOrder(most_recent_version):
+        msg = "{} version in root env is out-of-date.".format(name)
+    else:
+        return
+    if error_on_warn:
+        raise RuntimeError("{} Exiting.".format(msg))
+    else:
+        print(msg)
+
+
+def commit_changes(forge_file_directory, commit, cs_ver, cfp_ver):
+    msg = 'Re-rendered with conda-smithy {} and cf-pinning {}'.format(cs_ver, cfp_ver)
+    print(msg)
+
+    is_git_repo = os.path.exists(os.path.join(forge_file_directory, ".git"))
+    if is_git_repo:
+        has_staged_changes = subprocess.call(
+            [
+                "git", "diff", "--cached", "--quiet", "--exit-code"
+            ],
+            cwd=forge_file_directory
+        )
+        if has_staged_changes:
+            if commit:
+                git_args = [
+                    'git',
+                    'commit',
+                    '-m',
+                    'MNT: {}'.format(msg)
+                ]
+                if commit == "edit":
```

Comment:
So I kind of preferred edit to be the default. Not a big deal either way. Any reason in particular you think we should change?